torch/cuda/amp/grad_scaler.py

```diff
@@ -677,3 +694,30 @@ def _check_inf_per_device(self, optimizer: torch.optim.Optimizer) -> Dict[str, A
 
     def _found_inf_per_device(self, optimizer: torch.optim.Optimizer) -> Dict[str, Any]:
         return self._per_optimizer_states[id(optimizer)]["found_inf_per_device"]
+
+    def _update_scale_(
+        self,
+        _scale: torch.Tensor,
+        _growth_tracker: torch.Tensor,
+        found_inf_combined: torch.Tensor,
+    ):
+        torch._amp_update_scale_(
+            _scale,
+            _growth_tracker,
+            found_inf_combined,
+            self._growth_factor,
+            self._backoff_factor,
+            self._growth_interval,
+        )
+
+    def get_allow_fp16_grad(self) -> bool:
+        r"""Return a Python bool whether allow fp16 gradients"""
+        return self._allow_fp16_grad
+
+    def set_allow_fp16_grad(self, allow_fp16: bool) -> None:
```

Comment:
Why are we adding new public methods on `GradScaler`?

Reply:
I added self._allow_fp16 = False for GradScaler and True for ShardedGradScaler. If we do not need to expose this to user, then setter/getters removed or private

Reply:
I was wondering, if the code can support fp16 gradients, then is there a reason to have this flag? When would users want to error on fp16 gradients?

Reply:
I agree we should move them to private